[IV-25-26] Objetivo 1

vpedrosa opened the pull request:
Documentación del proyecto para el Objetivo 1:

- Historias de usuario
- User journeys
- Milestones

vpedrosa committed:
feat(objetivo-1): Entrega documentación proyecto

vpedrosa committed:
Merge branch 'dev' into Objetivo-1

vpedrosa committed:
Subo versión actualizada al nuevo problema

vpedrosa committed:
Subo modificaciones tras revisión

vpedrosa committed:
Subo versión más adaptada al objetivo-0

vpedrosa committed:
Reduzco las HUs a dos para que sean coherentes con GH y no planifiquemos de más

vpedrosa committed:
Modifico HUs para que se adapten mejor al milestone 0

vpedrosa committed:
Subo objetivo 1 completo

JJ commented:
Igual tampoco hacen falta tantos user journey...

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,31 @@
+
+Estructura de código con clases, interfaces y módulos que modelen el dominio del problema (condiciones de ola, criterios de surf, playas, períodos). El código debe seguir principios de DDD identificando correctamente entidades y objetos valor, aunque sin implementación funcional completa.
+
+# Milestone 1: Análisis de Períodos de Surf
```
Estás indicando un producto concreto. Qué desarrollo en concreto tiene que haber depende del desarrollo. No sabes si vas a tener el análisis ni si vas a tener otro PMV menos concreto.

vpedrosa replied:
Tienes razón. Voy a asignar explícitamente las HU001 y HU002 al Milestone 0. @JJ debería crear historias de usuario para todos los milestones? Las eliminé porque en una revisión me dijeron que aún no teníamos que hacer historias de usuario para todos los milestones, sólo para el primero.

vpedrosa replied:
Eliminaré las referencias a "clases", "interfaces", "DDD", "entidades" y "objetos valor". El entregable describirá el producto necesario para abordar las HUs asignadas, sin condicionar la implementación técnica.

vpedrosa replied:
Considerado y hecho.
Resuelvo este comentario.

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,31 @@
+
+# Milestone 1: Análisis de Períodos de Surf
+
+Implementación de la lógica que permite evaluar series de condiciones de ola e identificar períodos óptimos para surfear.
```
Igualmente, no sabes qué vas a hacer. Tampoco estás indicando con qué historia de usuario trabajas.

vpedrosa replied:
Aquí sigo con la pregunta un poco del comentario anterior: 

> @JJ debería crear historias de usuario para todos los milestones? Las eliminé porque en una revisión me dijeron que aún no teníamos que hacer historias de usuario para todos los milestones, sólo para el primero.

JJ commented on docs/milestones.md:
```diff
@@ -0,0 +1,31 @@
+
+## Entregable
+
+Producto que permita analizar condiciones de surf a lo largo del tiempo e identificar el período óptimo de 35-40 días.
```
El entregable no puede concretar la solución, sino las condiciones que hacen que esa solución sea correcta. Si identificas una solución concreta, estás condicionando el desarrollo.

vpedrosa replied:
Ok. Eliminaré la referencia específica a "35-40 días" del entregable. Describiré las condiciones que el producto debe cumplir sin especificar valores que condicionen la solución.

JJ reviewed (commented):
Además, tienes dos historias de usuario y no te refieres a ninguna de ellas en los milestone, ni secuencias cómo se van a abordar las mismas.
Sé que todas estas cosas probablemente no las necesitas por tu experiencia, pero te agradecería que trataras de entender el enfoque de la asignatura y vieras por qué todas las cosas son necesarias. Consulta también los errores comunes.
Te recuerdo además que el plazo de superación *de la convocatoria extraordinaria* termina el próximo miércoles. Por favor, revisa y vuelve a enviar con la mayor celeridad posible.

vpedrosa committed:
Refactoriza milestones eliminando condicionantes técnicos

Elimina referencias a soluciones técnicas específicas (DDD, clases, interfaces) y valores concretos que condicionan el desarrollo. Asigna HU001 y HU002 al Milestone 0 y reformula entregables para describir productos sin especificar implementación.

vpedrosa committed:
Restaura referencia a DDD como metodología válida

DDD es una metodología de trabajo que permite validar el producto verificando si se ha seguido, por lo que es apropiada para incluir en el milestone.

GabrielFranciscoSM reviewed (commented):
Hola, te he comentado las HUs. Creo que podría ayudar a que las planteras mejor que no limitases la(s) HU(s) a un único mileston, cómo haces implicitamente en el documento. Una HU puede estar asociada a más de una Milestone.

vpedrosa committed:
Simplifica estructura a una HU centrada en el problema del usuario

Unifica las HUs en una sola (HU1) que describe el problema real de Pablo como surfista: imposibilidad de identificar períodos óptimos por datos dispersos. Simplifica milestones a M0 (modelo) y M1 (lógica de negocio), ambos referenciando la HU1. Estructura adaptada al formato correcto de la asignatura.

GabrielFranciscoSM commented on docs/milestones.md:
```diff
@@ -0,0 +1,17 @@
+
+# [M1] Implementación de lógica de negocio
+
+Implementar la lógica de negocio necesaria para evaluar condiciones de surf e identificar períodos óptimos, resolviendo el problema descrito en la [HU1](user-stories.md#hu1-planificación-óptima-de-viajes-de-surf). La implementación debe ser verificable mediante testing.
```
¿No crees que evaluar condiciones de surf e identificar períodos óptimos es la solución al problema? Piensa en los diferentes objetivos (sobretodo el 2 y el 4) y en la metodología. Sólo siguiendo la metodología se podrá llegar a una solución correcta.

GabrielFranciscoSM commented on docs/milestones.md:
```diff
@@ -0,0 +1,17 @@
+
+## Entregable
+
+Código fuente funcional que permita analizar series de condiciones y determinar períodos óptimos de surf según criterios establecidos.
```
Igual que arriba

GabrielFranciscoSM reviewed (commented):
Sobre la HU, creo que está bastante mejor, y te he puesto un comentario para terminar de retocarla. Lo importante es no salirse del dominio del problema, y tratar de expresar la totalidad (o mayor parte) del dominio en la HU. Buen trabajo.

También te he comentado las milestones. Creo que reducir su número ha sido un acierto, y falta afinar las que ya tienes.

vpedrosa commented:
@GabrielFranciscoSM @JJ si podéis echarle un vistazo os lo agradezco, que se me agota el plazo 🥲​ Gracias!

vpedrosa committed:
Refactoriza HU1 y milestones para centrarse en el problema

Elimina referencias a soluciones específicas en la historia de usuario
y milestones, manteniendo el foco en el dominio del problema y dejando
que la solución emerja de seguir la metodología DDD.

GabrielFranciscoSM reviewed (approved):
Creo que has podido sistetizar toda la información que tenías en descripciones de milestones y HU adecuadas.

Te doy mi visto bueno, cuando quiera @JJ puede revisartelo también.

JJ reviewed (approved):
👍
